### DISCUSSION TITLE
How to add a new language

### DISCUSSION THREAD

**Discussion by yurtseverfazli** [idno/idno#3133] [General]
I have been trying to translate the source language to Turkish on Poedit. I created a tr_TR directory in /languages but it is not displayed on the site.

**Comment by cdn** [comment]
@yurtseverfazli currently Known uses the browser visiting it to determine which language to translate things to, provided those language files exist.

Does your browser have tr, tr-TR in its Accept Language configuration?

**Reply by yurtseverfazli** [reply]
I use google chrome as default browser, and the languages defined in its settings are Turkish, English (USA)   and English. I created a directory named tr_TR in "../languages", I used the language file in "/languages/source/" and I put the language translated in "../languages/tr_TR". 
That's all.

**Reply by cdn** [reply]
Sounds like the `.pot` - `.po` compilation step hasn’t happened?

**Reply by yurtseverfazli** [reply]
> Sounds like the `.pot` - `.po` compilation step hasn’t happened?

I use POEDIT to translate, and I save after translating. It sounds me that it doesn't see the "/language/tr_TR". I renamed the "../languages/en_US" as "../languages/xx_XX" and even the "../langages/source" and I renamed my tr_TR directory in "../languages" as "/en_US". But still it didn't change anything.

**Reply by cdn** [reply]
My mediocre Known extension [locale](https://github.com/cdn/locale) might assist in debugging?

**Reply by cdn** [reply]
Did you have the LC_MESSAGES subdirectory?
In either case of en_US or tr_TR, was your compiled file of translations `known.mo`?

**Comment by thierrymarianne** [comment]
Hello @yurtseverfazli,

As far as I can tell from pre-existing issues [on similar topics](https://github.com/idno/known/issues?q=language+is%3Aissue+is%3Aclosed),
the documentation has an entry entitled [`adding a translation for a language`](https://docs.withknown.com/en/latest/developers/languages/#adding-a-translation-for-a-language).

Please note that you'll need to run the [`build-lang`](https://github.com/idno/known/blob/dev/Gruntfile.js#L204) task with grunt,
once your `.POT` edition has been completed to your satisfaction.

You may also want to point out some of the usability difficulties you've encountered 
as an [issue](https://github.com/idno/known/issues/new/choose) by following some of the [contributing instructions](https://github.com/idno/known/blob/dev/CONTRIBUTING.md).

Thank you @cdn for pointing out the Accept Langage configuration, 
it didn't occur to me the browser could be configured in another language by default 😅.

**Reply by yurtseverfazli** [reply]
Thanks for your interest. I had read the lines above before I started translation from https://docs.withknown.com/en/latest/developers/languages/#adding-a-translation-for-a-language, but it couldn't help solve my problem. It looked so complicated. May be I am  a newbee for known. Whereas, I could translate some web cms using pot files. There must be an easier way to be able to do that.  Again thanks.

**Reply by thierrymarianne** [reply]
Do you see the expected translations when you edit the [source file](https://github.com/idno/known/blob/dev/languages/source/known.pot) and run the grunt task before accessing the web site again?

Do you mind sharing a fresh tree example with the translation files you've added,
so that we could perhaps try to figure together what didn't work out for you?
Later this week, I'll have perhaps some more time to try on my side as well.

**Reply by yurtseverfazli** [reply]
I have translated most of the strings, except 233 strings. I have shared it on my website usumda.com. You can on the address "https://usumda.com/file/view/3799/turkish-lang-files-for-the-known-social" and also shared below.
[tr_TR.zip](https://github.com/idno/known/files/9814604/tr_TR.zip)

**Comment by yurtseverfazli** [comment]
I have LC_MESSAGES subdirectory in only  en_US directory in langages.
The tree of the directory in root is below

Languages>en_US>LC_MESSAGES>.gitnore
                    >source>known.mo
                    >tr_TR>tr_TR.mo
                                  tr_TR.po

Language directory has three subdirectories as en_US which has LC_MESSAGES
subdirectory, source which has known.mo file and tr_TR which has two files
tr_TR.mo and tr_TR.po





18 Eki 2022 Sal 23:57 tarihinde cdn ***@***.***> şunu yazdı:

**Reply by cdn** [reply]
Known is expecting `/languages/xx_YY/LC_MESSAGES/known.mo`, so you would need `Languages>tr_TR>LC_MESSAGES>known.mo` which is your current `tr_TR.mo`

**Comment by yurtseverfazli** [comment]
As far as I see, I created the subdirectory LC_MESSAGES  in
../langages/tr_TR directory, then I noved tr_TR.mo into it. Nothing
changed. Evden I renamed the file, which I had moved into
../languages/tr_TR/LC_MESSAGES/, as known.mo , but sorry to say that
nothing changed.


19 Eki 2022 Çar 15:30 tarihinde cdn ***@***.***> şunu yazdı:

**Comment by yurtseverfazli** [comment]
I deleted the subdirectories en_US and source in order to force the system
to direct into tr_TR
But the language of the system kept the language in English.
So I think that the known doesn't use the files in ../languages.

19 Eki 2022 Çar 15:56 tarihinde Fazlı Yurtsever ***@***.***>
şunu yazdı:

**Reply by cdn** [reply]
something may be amiss with the `_($string, array $subs = [])` function in [`Idno/Core/Language.php`](https://github.com/idno/known/blob/dev/Idno/Core/Language.php)

**Comment by yurtseverfazli** [comment]
I think so. But I am not code writer, unfortunately.

19 Eki 2022 Çar 16:18 tarihinde cdn ***@***.***> şunu yazdı:

**Comment by cdn** [comment]
Semi–good news, the javascript that runs to update dates on recent posts is translating for Accept-Language; the most recent two posts on [my blog](https://definitely.cn) have Turkish strings…
